inbox/events/microsoft/events_provider.py

```diff
@@ -12,19 +12,31 @@
     MsGraphEvent,
     MsGraphSubscription,
 )
-from inbox.events.microsoft.parse import parse_calendar, parse_event
+from inbox.events.microsoft.parse import (
+    calculate_exception_and_canceled_occurrences,
+    parse_calendar,
+    parse_event,
+)
 from inbox.events.util import CalendarSyncResponse
 from inbox.models.account import Account
 from inbox.models.backends.outlook import MICROSOFT_CALENDAR_SCOPES
 from inbox.models.calendar import Calendar
-from inbox.models.event import Event
+from inbox.models.event import Event, RecurringEvent
 
 URL_PREFIX = config.get("API_URL", "")
 
 CALENDAR_LIST_WEBHOOK_URL = URL_PREFIX + "/w/microsoft/calendar_list_update/{}"
 EVENTS_LIST_WEBHOOK_URL = URL_PREFIX + "/w/microsoft/calendar_update/{}"
 
 
+# Microsoft Graph supports infinite and finite recursions.
+# By default recurring events are created 6 months into the
+# future but you can override it of course in the UI.
+# To prevent infinite or very long looping when searching
+# for exceptions and cancellations we need to establish a limit.
+MAX_RECURRING_EVENT_DURATION = datetime.timedelta(days=365)
```

Comment:
Maybe `MAX_RECURRING_EVENT_WINDOW`? Duration made me think it was how long the single event could be.